src/VaultSharp/V1/SecretsEngines/PKI/PKISecretsEngineProvider.cs

```diff
@@ -24,6 +24,17 @@ public async Task<Secret<CertificateCredentials>> GetCredentialsAsync(string pki
 
             return result;
         }
+        
+        public async Task<Secret<SignedCertificateData>> SignCertificateAsync(string pkiRoleName, SignCertificatesRequestOptions signCertificatesRequestOptions, string pkiBackendMountPoint = null, string wrapTimeToLive = null)
+        {
+            Checker.NotNull(pkiRoleName, "pkiRoleName");
+            Checker.NotNull(signCertificatesRequestOptions, "signCertificatesRequestOptions");
+
+            var result = await _polymath.MakeVaultApiRequest<Secret<SignedCertificateData>>(pkiBackendMountPoint ?? _polymath.VaultClientSettings.SecretsEngineMountPoints.PKI, "/sign/" + pkiRoleName, HttpMethod.Post, signCertificatesRequestOptions, wrapTimeToLive: wrapTimeToLive).ConfigureAwait(_polymath.VaultClientSettings.ContinueAsyncTasksOnCapturedContext);
+            result.Data.CertificateFormat = signCertificatesRequestOptions.CertificateFormat;
```

Comment:
needs a null check on result.Data

Reply:
It's assigned as in GetCredentialsAsync function. Thought that there will be some exception, error code from Api if there will be no data send. For my purpose i don't even need that CertificateFormat Assigned cause as a caller you already know it.